src/metabase/lib/metric/basics.cljc

```diff
@@ -0,0 +1,27 @@
+(ns metabase.lib.metric.basics
+  (:require
+   [metabase.lib.equality :as lib.equality]
+   [metabase.lib.join :as-alias lib.join]
+   [metabase.lib.metadata :as lib.metadata]
+   [metabase.lib.metadata.calculation :as lib.metadata.calculation]
+   [metabase.lib.query :as lib.query]
+   [metabase.lib.schema :as lib.schema]
+   [metabase.lib.schema.metadata :as lib.schema.metadata]
+   [metabase.lib.util :as lib.util]
+   [metabase.util.malli :as mu]))
+
+(defn source-metric
```

Comment:
I would like these named `source->metric` or `join-metric` for symmetry.

I think this function should take in a metadata-providable and a stage, then it could be called by `join->metric` as well (and could prove useful in qp)

Reply:
I've made the edits as suggested.